### ISSUE TITLE
Any plan about embed thumbnail decoding/encoding ? Like libheif

### ISSUE THREAD

**Issue by dreampiggy** [AOMediaCodec/libavif#127]
I'm not a expert at Image Compression format. But I guess since AVIF based on the HEIF container, it can store the embed thumbnail.

libheif's API: https://github.com/strukturag/libheif/blob/master/libheif/heif.h#L551-L565

**Comment by joedrago** [comment]
Yes, there is/was a loose plan involving `AVIF_DECODER_SOURCE_THUMBNAIL_ITEM`. I haven't implemented it yet simply because there hasn't been any demand. It would probably very easy to add read support, and I could probably offer a `thumbnail` avifImage on an avifEncoder that could be populated to insert one. 

I'll have to read the spec though, as the link you provided there suggests there could be any number of thumbnails, which is unfortunate.

**Comment by EwoutH** [comment]
I would love embedded thumbnails. Windows supports up to 256 pixels in the largest direction, that could be a good start.

When the image is smaller than 256x256 the image itself could also have the role of thumbnail.

**Comment by joedrago** [comment]
As a note, implementing thumbnail embedding into AVIF does not guarantee any kind of OS support. Microsoft's AVIF decoder still doesn't even honor YUV444 correctly, for example.

**Comment by novomesk** [comment]
When decoding thumbnail, I would like to do it without reading the entire file. When there is complete file in memory already, I think in majority of circumstances I would prefer decoding the full picture.

**Comment by gitoss** [comment]
> As a note, implementing thumbnail embedding into AVIF does not guarantee any kind of OS support. Microsoft's AVIF decoder still doesn't even honor YUV444 correctly, for example.

Concerning codecs: 

1. M$ support is annoying, it's not only limited but only available for rolling windows releases and not lts like 1809. Google released their own "WebP Codec for Windows 0.19.9" which works great, something similar would be useful for full avif support

https://storage.googleapis.com/downloads.webmproject.org/releases/webp/index.html

2. For anything qt-ish, an image plugin is required, I asked in the Windows QuickViewer git about this:

https://github.com/kanryu/quickviewer/issues/128
